### PR TITLE
Data Formats

### PR DESCRIPTION
**Pull request by jordan-smith721**
### At the docs level:

- Are the sections well organized? Is each topic where a reader would expect to find it?
- Are any topics missing?

### On each page:

- Are the sections well organized?
- Are all topics explained clearly and correctly?
- Is anything missing?
- Are there any upcoming changes to the driver that will affect the documentation?
- Are all code examples following best practices?

### CODE REVIEW

File: source/fundamentals/data-formats/bson.txt [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+
+.. literalinclude:: ../../includes/fundamentals/code-examples/Bson.cs
+   :start-after: start-bson-elements
+   :end-before: end-bson-elements
+   :language: csharp 
+   :copyable:
+   :dedent:
+
+The following code sample creates the same ``BsonDocument``, but each ``BsonElement``
+is a field-value pair:
```
[the remark this answers is not in the document]
Comment (jordan-smith721) [reply source/fundamentals/data-formats/bson.txt:64]: I think the intention here is just to make sure the reader is aware that there are two different ways to specify the same document - in case they come across it at some point. If it's not worth mentioning though I think we could probably just remove it.
@mongoKart feel free to correct me if I'm wrong on my reasoning here

File: source/fundamentals/data-formats/poco.txt [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+   }
+
+.. warning:: Multiple Id Fields
+
+   If you identify more than one property as the ``_id`` field using the
+   ``[BsonId()]`` attribute, the driver throws a
+   ``DuplicateBsonMemberMapAttributeException``. If your POCO
+   properties use more than one of the three recognized field
+   names (``Id``, ``id``, and ``_id``), the driver throws a
+   ``BsonSerializationException``.
```
[the remark this answers is not in the document]
Comment (jordan-smith721) [reply source/fundamentals/data-formats/poco.txt:201]: I think what this is saying is it throws an exception if you have some document like: 

`public class Person
  {
      public ObjectId Id { get; set; }
      public ObjectId _id { get; set; }
      ...
   }
`
@mongoKart I think you might be better suited to answer this one

File: source/fundamentals/data-formats/guid-serialization.txt [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+In this guide, you can learn how to serialize **globally unique identifiers**  
+(`GUIDs <https://learn.microsoft.com/en-us/dynamicsax-2012/developer/guids>`__),
+also known as **universally unique identifiers** (UUIDs).
+
+A GUID is a 16-byte integer that you can use as a unique ID for a MongoDB document. 
+Originally, GUIDs in MongoDB were represented as ``BsonBinaryData`` values of subtype 3. 
+Subtype 3 did not standardize the byte order during serialization, which led to 
+inconsistent serialization across MongoDB drivers. 
+To standardize the byte order and ensure consistent serialization across drivers, we 
+created ``BsonBinaryData`` subtype 4.
```
[the remark this answers is not in the document]
Comment (jordan-smith721) [reply source/fundamentals/data-formats/guid-serialization.txt:33]: I added a note towards the top of the page, let me know if that works!

File: source/fundamentals/data-formats/poco.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (mongoKart) [reply source/fundamentals/data-formats/poco.txt:201]: I think Jordan's interpretation is correct (Rea wrote the section on POCOs so I'm not certain). James, what exception cases do you think are likeliest to occur / most important to include here?

File: source/fundamentals/data-formats/bson.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (mongoKart) [reply source/fundamentals/data-formats/bson.txt:64]: That's right; users might see both syntaxes, and I also thought showing both sequentially would help them make the connection between `BsonElement` and a key-value pair. 

I'm in favor of showing the more idiomatic form and removing the `BsonElement` example.

[the remark this answers is not in the document]
Comment (JamesKovacs) [reply source/fundamentals/data-formats/bson.txt:64]: The dictionary initialization syntax is common in .NET. (e.g. `new SomeDict {{ key1, value1 }}`) Internally `new SomeDict {{ key1, value1}}` actually creates `KeyValuePair` objects just as we create `BsonElement` objects. Thus I think that `new BsonDocument {{ key1, value1 }}` should be obvious to most C# developers.

As for the compilation error, the `BsonArray` is initialized as:

```
new BsonArray {{ value1, value2 }}
```

But that is dictionary initialization syntax (taking key/value pairs), not array initialization syntax (taking a list of items). It should be:

```
new BsonArray { value1, value2 }
```

File: source/fundamentals/data-formats/poco.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (JamesKovacs) [reply source/fundamentals/data-formats/poco.txt:201]: Oh, I see. I would rephrase it as "If your POCO specifies the same database field multiple times (e.g. by specifying two different properties named `Id` and `_id`), then the driver throws a BsonSerializationException." I've honestly never seen this in the wild, but it is still worth pointing out.

File: source/includes/fundamentals/code-examples/poco.cs [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        Setup();
+        
+        // start-insert
+        var doc = new Clothing() 
+        { 
+            Name = "Denim Jacket",
+            InStock = false,
+            Price = 32.99m,
+            ColorSelection = new List<string>() {"dark wash", "light wash"}
+        };
```
[the remark this answers is not in the document]
Comment (JamesKovacs) [reply source/includes/fundamentals/code-examples/poco.cs:23]: Same for `Clothing`. Since you provide a property initializer list, you can omit the `()` after `new Clothing`. e.g. `new Clothing { Name = "Denim Jacket" }` rather than `new Clothing() { Name = "Denim Jacket" }`. The former is more idiomatic C#.

File: source/fundamentals/data-formats/guid-serialization.txt (hunk 3 above)
[the remark this answers is not in the document]
Comment (JamesKovacs) [reply source/fundamentals/data-formats/guid-serialization.txt:33]: Perfect.

**Review by JamesKovacs — CHANGES_REQUESTED** [review]
Minor changes requested.

File: source/fundamentals/data-formats/bson.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (mongoKart) [reply source/fundamentals/data-formats/bson.txt:64]: Makes sense. Let me know if my changes address these comments.

File: source/includes/fundamentals/code-examples/Clothing.cs [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    public ObjectId Id { get; set; }
+
+    [BsonElement("name")]
+    public string Name { get; set; }
+
+    [BsonElement("inStock")]
+    public bool InStock { get; set; }
+
+    [BsonElement("price")]
+    [BsonRepresentation(BsonType.Double)]
```
Comment (JamesKovacs) [inline source/includes/fundamentals/code-examples/Clothing.cs:18]: Why don't we use `BsonType.Decimal128` here?

**Review by JamesKovacs — CHANGES_REQUESTED** [review]
Minor suggestion...

File: source/includes/fundamentals/code-examples/Clothing.cs (hunk 5 above)
Comment (mongoKart) [reply source/includes/fundamentals/code-examples/Clothing.cs:18]: I agree. I've always used Decimal for financial values. I'll make that change.

**Review by JamesKovacs — APPROVED** [review]
LGTM
